nomad/structs/services.go

```diff
@@ -458,6 +461,10 @@ func (s *Service) Hash(allocID, taskName string, canary bool) string {
 	for _, tag := range s.CanaryTags {
 		io.WriteString(h, tag)
 	}
+	for k, v := range s.Meta {
```

Comment:
I don't think we want meta as part of the hash. This is used for generating the service ID so if you changed the meta it would cause the service to be reregistered which doesn't seem like what we want.

Reply:
I would probably use meta for stuff I currently use tags for today since it would be proper KV map. This includes ~dynamic things from the Nomad client itself. Would they still be re-synced on change without having them part of the hash?

Reply:
I reread through the code, I had thought the answer to that question @jippi would be yes, but I was wrong!

So it looks like we definitely want to keep the meta as part of the hash. I thought it was still part of the service ID, but that was fixed in 0.9. 

I think we want to do more than just writting the k/vs to the hash though. For example `{"foo":"bar"}` and `{"f":"oo", "b":"ar"}` would hash to be the same in this cause. While probably not a big deal lets play it safe with something like:
```
  if len(s.Meta) > 0 {
		fmt.Fprintf(h, "%v", s.Meta)
	}
```